src/wstore/store_commons/middleware.py

```diff
@@ -146,18 +146,18 @@ def get_api_user(request):
 
     # Check if the user already exist
     try:
-        user = User.objects.get(username=nick_name)
+        user = User.objects.get(username=user_name)
     except:
-        user = User.objects.create(username=nick_name)
-
-    # Update user info
-    user.email = email
-    user.userprofile.access_token = token_info[1]
-    user.userprofile.complete_name = display_name
-    user.userprofile.actor_id = nick_name
-
-    user.is_staff = settings.ADMIN_ROLE.lower() in roles
-
+        user = User.objects.create(username=user_name)
+
+    if nick_name == user_name:
+        # Update user info
+        user.email = email
+        user.userprofile.access_token = token_info[1]
```

Comment:
You still need to save the access token in the userprofile even if you are acting as an organization. Take into account that access tokens are given at user level